src/components/CodeInput/index.js

```diff
@@ -0,0 +1,142 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier } from '../../libs/hooks';
+import { getNormalizedValue, getValidValue } from './helpers';
+import HelpText from '../Input/styled/helpText';
+import InputItems from './inputItems';
+import Label from '../Input/label';
+import RenderIf from '../RenderIf';
+import StyledFieldset from './styled';
+import StyledTextError from '../Input/styled/errorText';
+
+/**
+ * The CodeInput is an element that allows to fill a list of numbers, suitable for code validations.
+ * @category Form
+ */
+function CodeInput(props) {
+    const {
+        id,
+        name,
+        value,
+        label,
+        bottomHelpText,
+        codeLength,
+        disabled,
+        required,
+        readOnly,
+        error,
+        tabIndex,
+        onClick,
+        onChange,
+        onFocus,
+        onBlur,
+        onKeyDown,
+        className,
+        style,
+    } = props;
+
+    const valueProp = getValidValue(value, codeLength);
+    const inputId = useUniqueIdentifier('code-input');
+    const inputRef = useRef();
+
+    const handleOnChange = (inputValue, inputIndex) => {
+        onChange(getNormalizedValue(inputValue, inputIndex, valueProp, codeLength));
+    };
+
+    return (
+        <StyledFieldset className={className} name={name} style={style} id={id}>
+            <Label
+                label={label}
+                hideLabel={!label}
+                required={required}
+                readOnly={readOnly}
+                inputId={inputId}
+            />
```

Comment:
since this is a group of inputs wrapped by a fieldset then here we need to use a legend tag instead of label then we can't reuse the Label component